eclipse-collections-api/src/main/java/org/eclipse/collections/api/collection/FixedSizeCollection.java

```diff
@@ -23,79 +23,6 @@
 public interface FixedSizeCollection<T>
         extends MutableCollection<T>
 {
-    /**
```

Comment:
@aboullaite I'd like to understand why was this javadoc removed. This indicates the behavior for a `FixedSizedCollection` is different than other `MutableCollection` instances. This is an awkward behavior here to be sure, but this is the unfortunate nature of something that is fixed in size.

Reply:
@aboullaite Now that I have taken a look, please push this JavaDoc down to the implementation classes so it is not lost. The behavior is definitely different out of necessity. I will try and clean this up a bit after we merge your PR.

Reply:
I disagree that the Javadoc should be pushed down. It's on an interface where it probably belongs, and the fact that it's abstract is good. Subclasses must have an interesting implementation. I'd suggest just revert the change to FixedSized* interfaces.

Good catch @donraab

Reply:
The only issue wit this is that the implementations can't be left abstract because they are already defined as default methods in a parent interface.

Reply:
Are we talking about in the interface (FixedSizeCollection) or in the concrete classes that implement it? It's ok to override a default interface method with an abstract interface method. And before we started FixedSizeCollection's methods were abstract, so I know the concrete classes implement these methods. I'm not understanding the problem here.

Reply:
In the interface.

> It's ok to override a default interface method with an abstract interface method.

My bad. I thought I remembered having an issue here overriding a default method with an abstract method in an interface. Just did some googling and this seems to be ok. If this isn't required then the interface can just be reverted.

Reply:
@donraab @motlin The compiler complains about having an abstract and a default method with matching signatures. The problem happens in `FixedSizeSet` and `FixedSizeList` since both `MutableSet` and `FixedSizeCollection` define  methods `with()`, `without()` ... and one is default and the other is abstract, the compiler requires the subinterface to explicitly declare which one it should inherit by overriding the methods again.

Should I override the methods in sub-interfaces (FixedSizeSet, FixedSizeList) , or remove the the abstract methods from `FixedSizeCollection` and push down javadoc documentation to implementation classes